engine/common/provider/engine.go

```diff
@@ -224,3 +250,66 @@ func (e *Engine) onEntityRequest(originID flow.Identifier, req *messages.EntityR
 
 	return nil
 }
+
+func (e *Engine) processQueuedRequestsShovellerWorker(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+	ready()
+
+	e.log.Debug().Msg("process entity request shoveller worker started")
+
+	for {
+		select {
+		case <-e.requestHandler.GetNotifier():
+			// there is at least a single request in the queue, so we try to process it.
+			e.shovelEntityRequests()
```

Comment:
I find this terminology confusing. Elsewhere we use `processAvailableMessages`. Can you switch to that?

Reply:
https://github.com/onflow/flow-go/pull/3404/commits/ea22a1c4bf9cb025e44bd71773528ad4ce7c407c